### PR TITLE
add PlainTime basics

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by MaxGraey** [comment]
@ColinEberhardt ready for review

**Comment by MaxGraey** [comment]
@ColinEberhardt ready for review

File: assembly/plaindate.ts [hunk 1]
```diff
@@ -47,16 +52,26 @@ export class PlainDate {
...
       }
+      throw new TypeError("invalid date type");
     }
-    throw new TypeError("invalid date type");
   }
 
-  constructor(readonly year: i32, readonly month: i32, readonly day: i32) {}
+  constructor(readonly year: i32, readonly month: i32, readonly day: i32) {
+    // if (!(
+    //   checkRange(month, 1, 12) &&
```
Comment (MaxGraey) [inline assembly/plaindate.ts:84]: @ColinEberhardt that's pretty strange. After uncomment this simple check: 
`if (!checkRange(month, 1, 12)) throw error`

test 
```it(`Temporal.PlainDate.from(${date}) is not the same object)`, () => notEqual(PlainDate.from(date), date));```
will fail.

**Comment by ColinEberhardt** [comment]
Thanks  @MaxGraey looks good, and thanks for making a start on PlainTime. As part of  this PR can you also add the unit tests from https://github.com/tc39/proposal-temporal/blob/main/polyfill/test/plaintime.mjs so that we can track progress?

**Comment by MaxGraey** [comment]
@ColinEberhardt done!

Also I fixed potentially issue:
https://github.com/ColinEberhardt/assemblyscript-temporal/pull/8/files#diff-41d84c30242d77685ddc03e58098387cd75861eca7cae8bb192c0680631373d4L133

**Comment by ColinEberhardt** [comment]
Looks good to me - thanks. I'll add a README to the tests folder shortly that outlines the overall test approach so that we can maintain consistency.

**Pull request merged**